flow/util/misc.py

```diff
@@ -251,3 +251,50 @@ def _cached_partial(func, *args, maxsize=None, **kwargs):
 
     """
     return lru_cache(maxsize=maxsize)(partial(func, *args, **kwargs))
+
+
+class _bidict(dict):
```

Comment:
Is this private because we don't want users to play around with this class?

Reply:
Yes, I don't want to make this part of the public API. If it's public, it's harder to make breaking changes or remove it if it's not being used anymore.